Add 2023-09-06 agenda and minutes (#567)
solid/specification: meetings/2023-09-06.md

* SC: We'll have some follow-ups next.
* SC: I'd like TBL to comment on Vision and RW on Legal. 
* TBL: On licensing, I posted my conclusion on Matrix. If the charter is finalized that is incompatible with you wanting my review. In particular snippets of code you want to paste in your program. We looked at the MIT license but then looked at MDN, and they say that any snippet wherever it is, will be public domain. I think the MDN people had a good idea. That discussion obviously did not make its way into the charter.
* WT: Thanks TBL for the input. Discussion on the GH issue was about MIT license, which is what was incorporated into the charter, but I think it is definitely possible to add public domain for snippets. 
* TBL: Happy with MIT license. But CC0 for code snippets in specs or tutorials. 
* SC: I think it's a work item.
* TBL: No, part of a work item. Code snippets within a document that could have W3C/MIT license, but snippets are CC0. 
* SC: This licensing issue is about the work item, you are saying distinguishing snippets. That was one of the suggestions in the issue. 

> * RW: I want to reassure TBL that the only issue I see is the notices. Imagine you have software and need notices. We already have a practice of notices, you can take the code somewhere, but then a multitude of notices. IF we accept that notices should be provided, then the CLA would already do what Tim wants. You can co-license with MIT because CLA says "I grant perpetual license with no obligation to reproduce, distribute, etc.." meaning we have it already. This is rather later for WG, when you want to clearly distinguish the specification with the rules where you don't want someone making derivatives of the spec, but you want the code/pseudo code to be reused as much as possible. Similar issue to test-suite, where you don't want to ??? As it stands, the things TBL wants to accomplish can be done now. From a legal standpoint we are clear. Clarify pseudocode is MIT license. 

Reviewer: ```suggestion
* RW: I want to reassure TBL that the only issue I see is the notices. Imagine you have software and need notices. We already have a practice of notices, you can take the code somewhere, but then a multitude of notices. IF we accept that notices should be provided, then the CLA would already do what Tim wants. You can co-license with MIT because CLA says "I grant perpetual license with no obligation to reproduce, distribute, etc." meaning we have it already. This is rather late for WG, when you want to clearly distinguish the specification with the rules where you don't want someone making derivatives of the spec, but you want the code/pseudocode to be reused as much as possible. Similar issue to test-suite, where you don't want to ??? As it stands, the things TBL wants to accomplish can be done now. From a legal standpoint we are clear. Clarify pseudocode is MIT license. 
```

> * WT: As I understood Tim, he wanted to be able to copy code snippets from specs without the need to also copy a copyright notice into the derivative work. Even snippets in the spec follows MIT, the notice needs to be included. If we want to be able to copy snippet and no notice we need ??? My question to RW is once we go from CG to WG, our specifications become W3C document. W3C license says snippets fall under W3C software license. As we transition, how can this be compatible with snippets being public domain? 

Reviewer: Includes @woutermont's fix
```suggestion
* WT: As I understood Tim, he wanted to be able to copy code snippets from specs without the need to also copy a copyright notice into the derivative work. Even snippets in the spec follows MIT; the notice needs to be included. If we want to be able to copy snippet and need no notice, we need a public domain license. My question to RW is: 
  Once we go from CG to WG, our specifications become W3C documents. W3C license says snippets fall under W3C software license. As we transition, how can this be compatible with snippets being public domain?
```

> * RW: We allow specification to have a ??? under W3C license. I'd insist we keep ??? and have a disclaimer. The only thing we need to take is notification, which is a nuisance. It's no issue to transition even if you put code snippets as being public domain. I want to prevent some big corp coming, taking the work, making incompatible changes and claiming they can. What Tim wants is being able to take snippets and put in software with no hassle, but we still should be able to say something is not compatible with what the group consensus was. You have to include an adapted MIT license, I will help you do so. In transitioning to WG, I don't see any issue.

Reviewer: ```suggestion
* RW: We allow specifications to have a ??? under W3C license. I'd insist we keep ??? and have a disclaimer. The only thing we need to take out is notification, which is a nuisance. It's no issue to transition, even if you put code snippets as being public domain. I want to prevent some big corp coming, taking the work, making incompatible changes, and claiming they can. What Tim wants is to be able to take snippets and put them in software with no hassle, but we still should be able to say that something is not compatible with what the group consensus was. You have to include an adapted MIT license; I will help you do so. In transitioning to WG, I don't see any issue.
```

* WT: Even if W3C licensing document says code snippets must be under software license, we still can change that license?
* RW: Yes.
* SC: How do we go from what we have in the charter right now to finalizing, merging charter after making clarification? Does anyone have any objections? Especially TBL. Does what RW suggests work for you? 
* RW: To reiterate: the plan is to make a W3C software license with no notification but a disclaimer.
* TBL: Happy to ??? but only for the code snippets that the notification is removed. I want what the WG produces to be usable by people. 
* WT: RW, does what we currently have in charter accomplish what you suggest?

> * RW: What is in the charter is not in the document and the object of the rule is not clear. In the charter we can give the precise URL of a specification we can make assertions. Having MIT does not do the trick because of notification. I suggest we put W3C software license and in the annex say the code snippets are under W3C license without application of notification. I may want to create a W3C snippet license that does exactly what Tim wants but this will take a long time. There's no issue whatsoever with spec authors ??? I can give you a 3-liner for the charter so contributors commit to this. 

Reviewer: ```suggestion
* RW: What is in the charter is not in the document, and the object of the rule is not clear. In the charter, we can give the precise URL of a specification; we can make assertions. Having MIT does not do the trick because of notification. I suggest we put W3C software license, and, in the annex, say the code snippets are under W3C license without application of notification. I may want to create a W3C snippet license that does exactly what Tim wants but this will take a long time. There's no issue whatsoever with spec authors ??? I can give you a 3-liner for the charter so contributors can commit to this. 
```

> * WT: I agree we should write whatever license we use in the documents. My question is given the text of the charter, which informs the line that should be in the documents, can we simply switch MIT to "W3C software without notice"? 

Reviewer: ```suggestion
* WT: I agree we should write whatever license we use in the documents. My question is:
  Given the text of the charter, which informs the line that should be in the documents, can we simply switch MIT to W3C software, without notice? 
```

* RW: Yes. 

> * RG: My question to RW is, is it already the case in copyright law that if you take a small portion of the work and you're doing it in a reasonable way you don't even need to worry about copyright because you're not using it as is? 
> * RW: If you never travel to Europe in your life you could operate this way. In the US you have fair use sometimes. 

Reviewer: ```suggestion
* RG: My question to RW is:
  Is it already the case in copyright law that, if you take a small portion of the work and you're doing it in a reasonable way, you don't even need to worry about copyright because you're not using it as is? 
* RW: If you never travel to Europe in your life, you could operate this way. In the US, you have fair use, sometimes. 
```

* TBL: But it's hairy.

> * RW: In Europe your code does not need to be ??? I would not spare those 3 lines that we have to write to save hassle for people. 
> * SC: Pre-proposal: replacing MIT license with W3C software without notice. We can come back to making the text clear in the specs themselves. 

Reviewer: ```suggestion
* RW: In Europe, your code does not need to be ??? I would not spare those 3 lines that we have to write to save hassle for people. 
* SC: Pre-proposal: replace MIT license with W3C software without notice. We can come back to make the text clear in the specs themselves. 
```

* TBL: Good to have the intention in the charter. 
* SC: Let's make that change request in the PR. Agreement is to merge the charter PR. 

> * RW: Main issue remaining is, if people contribute and commit to the CLA it requires notification so at some point we can have a web form where people give up the notification for those who originally contributed text to the spec. When is it recharterting? Current charter, when does it apply? IS it in effect?

Reviewer: ```suggestion
* RW: Main issue remaining is:
  If people contribute and commit to the CLA, it requires notification, so at some point we can have a web form where people give up the notification, for those who originally contributed text to the spec. When is it rechartering? Current charter, when does it apply? Is it in effect?
```

* SC: It was effective September 1st. We're talking about the CG charter. First charter. Amendments can come if we want to make changes.

> * RW: How can I get evidence of the commitment? The intended result is clear. Probability that somebody comes and says they did not commit is close to zero, but to do it correctly we must record commitment. 

Reviewer: ```suggestion
* RW: How can I get evidence of the commitment? The intended result is clear. Probability that somebody comes and says they did not commit is close to zero, but to do it correctly, we must record commitment. 
```

* WT: Does this differ from the process where we get a FSA? 
* RW: We can do that on the same step. 
* SC: That's on the CG page somewhere. There's a way to tell who made a commitment and who didn't. 
* SC: Objections to making the change to W3C license without notice and following up with specific text to introduce in text for next releases? 
* SC: We'll merge the charter and if we find an error we can have another discussion and amend. 
* WT: +1
* TBL: +1
* eP: +1

DECISION: apply suggested change, SC to merge. 

> * eP: Rigo mentioned something about test suite. Does the charter already address or do we need to do something special?
> * RW: Test suite is double licenced, you don't need to do anything. If you make claims of conformance you cannot alter. 

Reviewer: ```suggestion
* eP: Rigo mentioned something about test suite. Does the charter already address, or do we need to do something special?
* RW: Test suite is double licensed; you don't need to do anything. If you make claims of conformance, you cannot alter. 
```

> * eP: For test suite it is better not to use MIT but the one recommended by W3C. 

Reviewer: ```suggestion
* eP: For test suite, it is better not to use MIT, but the one recommended by W3C. 
```

* RW: Not very different. For the test suite, for claims of conformance or intend to use it for claims of conformance, you cannot take altered code, because then your claims would be false. Copyright is an additional kind of helper, not main target. 

> * WT: The sentences in the charter about this is exactly what W3C has about double licensing, so we don't deviate from that. 

Reviewer: ```suggestion
* WT: The sentences in the charter about this are exactly what W3C has about double licensing, so we don't deviate from that. 
```

* SC: RW, in the license section we have to "note that". Do you recommend that we keep that text in or remove? 

> * RW: First of all you say WG instead of CG which is wrong. I think per W3C process this is a community process so normal process doesn't apply. I don't thin kit makes a legal different but would look nicer if you note that artifacts that become part of this group's TR will fall under W3C software license.
> * WT: The notes are meant to indicate that independent of the license we use, these items will be under a WG's license. 

Reviewer: ```suggestion
* RW: First of all, you say WG instead of CG, which is wrong. I think, per W3C process, this is a community process, so normal process doesn't apply. I don't think it makes a legal different, but would look nicer if you note that artifacts that become part of this group's TR will fall under W3C software license.
* WT: The notes are meant to indicate that, independent of the license we use, these items will be under a WG's license. 
```

> * RW: You can leave that to the WG. I would keep the paragraph but make it so that it reads as "the intention is to..." because whether it happens or not we will decide later.

Reviewer: ```suggestion
* RW: You can leave that to the WG. I would keep the paragraph, but make it so that it reads as "the intention is to...", because whether it happens or not, we will decide later.
```

* SC: Can we add a revision to that before we merge? WT. 
* WT: Yes. 
* eP: In the next 5 minutes?
* SC: Yes.
* RW: You don't have to use test suite license for tests. 
* SC: Can we draft in the minutes now? 
* RG: Can I ask RW to put the recommendations somewhere here we can follow?
* RW: I can create a pull request.
* RG: Appreciate it.
* WT: "Note that the intention of this group is to propose its documents as Technical Reports to a Working Group, where the published documents will fall under the [W3C Document License]."
* SC: Works for me.
* RW: Completely fine.
* WT: "Note that the intention of this group is to propose its software artifacts and code snippets included in documents to a Working Group's Technical Reports, where they will fall under the [W3C Software License] without notification obligation."
* RW: That's fine, exactly what I wanted.
* WT: "In accordance with the CLA, the group licenses all its software artifacts under the [W3C Software license], but without notification obligation."
* RW: Fine. The important part is distinction between document that has rules and software artifacts. 

> * eP: Software artifacts, only snippets or something other than test suites? 
> * RW: Here we talk about the rules, understanding software snippets that are written in the spec: you have a rule "xyz, software must do abc, here is an example". When you have other software you want to put under W3C you have a free choice of license. My favorite is Apache because it allows commercial usage but offers patent protection. 

Reviewer: ```suggestion
* eP: Software artifacts: only snippets, or something other than test suites? 
* RW: Here we talk about the rules, understanding software snippets that are written in the spec: you have a rule "xyz, software must do abc, here is an example". When you have other software you want to put under W3C, you have a free choice of license. My favorite is Apache, because it allows commercial usage but offers patent protection. 
```
